docs/tutorial_langcheckchat.md

```diff
@@ -0,0 +1,125 @@
+- Step 1: retrieve information from some data source and insert it into an LLM’s context
+- Step 2: query the LLM
+
+The main benefits of this approach vs. simply querying the LLM directly are:
+- You can give the LLM access to outside data sources, such as your organization’s private data
+- You can have a higher degree of control over the LLM’s behavior by enforcing it to only answer queries based on the data source you provide it
+
+Because of these benefits, RAG is particularly well suited for applications such as querying your organization’s private data, or for applications where you want the LLM to be grounded to some specific data source.
+
+## Building the initial RAG system
+
+For this tutorial, we will use [LlamaIndex](https://github.com/run-llama/llama_index) as the framework to build the RAG system, which will call the OpenAI API as the base LLM model under the hood. LlamaIndex is an excellent framework to quickly spin up your own simple RAG system, but also has many advanced features if you want to further improve your system’s performance down the line.
+
+
+The first thing we need to do is ***load*** our data, which in our case is a list of documentation web pages for LangCheck. In LlamaIndex, we can load data like this:
+
+```python
+from llama_index.readers import SimpleWebPageReader, StringIterableReader
+
+
+# SimpleWebPageReader reads the text on the web page, turning html into
+# equivalent Markdown structured text
+loader = SimpleWebPageReader(html_to_text=True)
+pages = [<list of LangCheck documentation pages>]
+documents = loader.load_data(urls=pages)
+```
+
+Next, we want to ***index*** the data, meaning we want to structure the data in a way that it is queryable. For RAG, the most popular way to index data is to generate vector embeddings and store them in a vector store. In our case, we will generate the embeddings using OpenAI, so we first need to set the OpenAI API key (if you don’t have an OpenAI API key, you can create one by signing up [here](https://platform.openai.com/account/api-keys)).
+
+```python
+import os
+from llama_index import VectorStoreIndex
+
+
+os.environ['OPENAI_API_KEY'] = 'YOUR_OPENAI_API_KEY'
+
+
+# OpenAI's "text-embedding-ada-002" is used as the embedding model by default
+index = VectorStoreIndex.from_documents(documents)
+```
+
+Finally, we can ***query*** the system and get a response! LlamaIndex lets us do this with just one line of code.
+
+```python
+# OpenAI's "gpt-3.5-turbo" model is used as the LLM by default
+response = index.as_query_engine().query("How can I install langcheck?")
+print(response)
+```
+
+Here’s the response from the LLM:
+> To install LangCheck, you can run the following command:
+>
+> pip install langcheck
+>
+> Please note that LangCheck requires Python 3.8 or higher to work properly.
+
+We can also see the sources that were retrieved from the index. By default, the top 2 most relevant source nodes are returned, which is what we see in `response.source_nodes`.
+
```

Comment:
Maybe add sample code here to explain how to use LangCheck to measure ai_disclaimer and factual_consistency?

Reply:
I added a bunch of sample code in https://github.com/citadel-ai/langcheck/pull/95/commits/dc588e55d0c9d1b39fce89a31c4a163d76ef16ee, ptal!